src/vm/database/marf.rs

```diff
@@ -210,17 +210,27 @@ impl MarfedKV {
     ///   ClarityDatabase or AnalysisDatabase -- this is done at the backing store level.
 
     pub fn begin(&mut self, current: &StacksBlockId, next: &StacksBlockId) {
-        self.marf.begin(current, next)
-            .expect(&format!("ERROR: Failed to begin new MARF block {} - {})", current, next));
+        {
+            let mut tx = self.marf.begin_tx()
+                .expect(&format!("ERROR: Failed to begin new MARF block {} - {})", current, next));
+            tx.begin(current, next)
+                .expect(&format!("ERROR: Failed to begin new MARF block {} - {})", current, next));
+            tx.commit_tx();
```

Comment:
I'm not sure to understand the begin / commit (and nothing between) in this scope, can you please hint me? Thanks!

Reply:
Yeah -- so this is definitely a hack to allow the Clarity VM to continue to interact with the MARF as it did previously. Previously, the way this function worked is that when it called `self.marf.begin`, internally the MARF would open the new trie inside of a SQL transaction and then commit it. With this PR, the _caller_ of the MARF functions is responsible for managing the SQL transaction state, so this function, in order to behave as it did before, needs to start and commit the SQL transaction even though the MARF trie isn't written yet (the thing that gets committed is just the trie lock).